### PR TITLE
Add TCF 2.3 Disclosed Vendors segment support

### PR DESCRIPTION
**Pull request by ggierak**
## Summary

- Add support for parsing the **Disclosed Vendors** segment (SegmentType=1) introduced in TCF 2.3
- Expose two new methods on the `VendorConsents` interface: `MaxDisclosedVendorID()` and `DisclosedVendor(id)`
- Maintain full backward compatibility with TCF 2.2 strings (methods return `0` / `false` when the segment is absent)

## Details

TCF 2.3 makes the Disclosed Vendors segment mandatory in TC Strings:

```
[Core String].[Disclosed Vendors].[Publisher TC]
```

This segment records which vendors were disclosed to the user by the CMP. Previously, `ParseString()` discarded all segments after the first `.` separator. Now it decodes each additional segment, identifies its type via the first 3 bits, and parses Disclosed Vendors (type `1`) using the same bitfield/range encoding as vendor consent.

### Changes

| File | Change |
|------|--------|
| `api/consent.go` | Add `MaxDisclosedVendorID()` and `DisclosedVendor(id)` to `VendorConsents` interface |
| `vendorconsent/tcf2/consent.go` | Parse additional segments in `ParseString()`, add `parseDisclosedVendors()` and `parseSegmentRangeSection()` |
| `vendorconsent/tcf2/metadata.go` | Add `disclosedVendors` field and accessor methods with nil-safe handling |
| `vendorconsent/tcf1/metadata.go` | Stub implementations returning zero-values |
| `vendorconsent/tcf2/consent_test.go` | Tests for TCF 2.3 parsing, TCF 2.2 backward compat, and edge cases |

## Test plan

- [x] Parse TCF 2.3 string with range-encoded Disclosed Vendors (vendors 1-5, 100, 404)
- [x] Parse TCF 2.2 string without Disclosed Vendors segment (returns 0/false)
- [x] Edge cases: vendor ID 0 and beyond MaxDisclosedVendorID
- [x] Existing tests pass (`go test ./...`)

### CODE REVIEW

**Comment by osazos** [comment]
Hi @ggierak, thanks for working on this. There is already an open PR addressing the same issue (#48), and it’s currently under review and close to being ready (see https://github.com/onfocusio/go-gdpr/pull/1).
Would you mind joining the discussion there and contributing to that PR so we can avoid duplicating efforts?

File: vendorconsent/tcf2/consent_test.go [hunk 1]
```diff
@@ -15,3 +15,52 @@ func TestParseLegitIntSetWithRangeSection(t *testing.T) {
...
+	consent, err := ParseString("CQSbk4AQSbk4ANwAAAENAwCgAAAAAAAAAAYgACPAAAAA.IDKQA4AAgAKAGQAygAAA.YAAAAAAAAAAA")
+	assertNilError(t, err)
+
+	metadata := consent.(ConsentMetadata)
+
+	// Vendor ID 0 should return false
+	assertBoolsEqual(t, false, metadata.DisclosedVendor(0))
+	// Vendor ID beyond MaxDisclosedVendorID should return false
+	assertBoolsEqual(t, false, metadata.DisclosedVendor(405))
+}
```
Comment (anishb-ms) [inline vendorconsent/tcf2/consent_test.go:66]: bitfield-encoded Disclosed Vendors is untested. All three new tests use range-encoded segments (IsRangeEncoding=1). The else branch in parseDisclosedVendors (consent.go:148-150) that calls parseBitField has no coverage. Worth adding one bitfield-encoded case.
